web/packages/teleterm/src/mainProcess/agentRunner/agentRunner.ts

```diff
@@ -21,12 +21,15 @@ import stripAnsi from 'strip-ansi';
 
 import Logger from 'teleterm/logger';
 import { RootClusterUri } from 'teleterm/ui/uri';
+import { createFileLoggerService, LoggerColor } from 'teleterm/services/logger';
 
 import { generateAgentConfigPaths } from '../createAgentConfigFile';
 import { AgentProcessState, RuntimeSettings } from '../types';
 import { terminateWithTimeout } from '../terminateWithTimeout';
 
 const MAX_STDERR_LINES = 10;
+// https://github.com/gravitational/teleport/blob/1212306cff9be443286cdf0e8cbdf6471fc392d7/lib/service/signals.go#L59
+const AGENT_GRACEFUL_SHUTDOWN_SIGNAL = 'SIGQUIT';
```

Comment:
I'm not sure if we should use SIGQUIT. It makes it so that if there are any SSH connections active, the agent is going to wait for them to end before exiting. So any time you stop the agent while a session is running, you'll see this:

![exited with sigkill](https://github.com/gravitational/teleport/assets/27113/ad453927-bff5-448f-9e5f-77191aeaf1fa)

This would make sense if we notified the user about this and offered them a way to close the sessions, but what's happens now is we simply wait for 5 more seconds and then send SIGKILL.

SIGTERM should be fine here IMHO.

Reply:
Uh, I didn't think `SIGQUIT` would be that graceful... Thanks for catching this, I'm reverting it.

Reply:
The only reason I knew about is is because of the work on the agent cleanup daemon.